python/ml4ir/base/model/optimizer.py

```diff
@@ -37,38 +24,83 @@ def get_optimizer(
     References:
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
+        https://arxiv.org/pdf/1506.01186.pdf
 
     FIXME:
         Define all arguments overriding tensorflow defaults in a separate file
         for visibility with ml4ir users
     """
-    # Define an exponential learning rate decay schedule
-    learning_rate_schedule = ExponentialDecay(
-        learning_rate,
-        decay_steps=learning_rate_decay_steps,
-        decay_rate=learning_rate_decay,
-        staircase=True,
-    )
+    model_config = file_io.read_yaml(model_config_file)
 
-    if optimizer_key == OptimizerKey.ADAM:
-        return tf_optimizers.Adam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.NADAM:
-        return tf_optimizers.Nadam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.ADAGRAD:
-        return tf_optimizers.Adagrad(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.SGD:
-        return tf_optimizers.SGD(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.RMS_PROP:
-        return tf_optimizers.RMSprop(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
+    if 'lr_schedule' not in model_config:
+        learning_rate_schedule = ExponentialDecay(
+            initial_learning_rate=0.01,
+            decay_steps=10000000,
+            decay_rate=1.0,
+            staircase=True,
         )
+
+    else:
+        lr_schedule = model_config['lr_schedule']
+        lr_schedule_key = lr_schedule['key']
+        if lr_schedule_key == LearningRateScheduleKey.EXPONENTIAL:
+            learning_rate_schedule = ExponentialDecay(
+                initial_learning_rate=lr_schedule['learning_rate'] if 'learning_rate' in lr_schedule else 0.01,
```

Comment:
Another important aspect are the hard-coded default values here.. We should have a separate config file that controls them, unless we have profound evidence that these values work well in practise. But this is for another PR!